conan/internal/rest/remote_manager.py

```diff
@@ -243,10 +248,23 @@ def get_latest_package_reference(self, pref, remote, info=None) -> PkgReference:
 
         cached_method = remote._caching.setdefault("get_latest_package_reference", {})
         try:
-            return cached_method[pref]
+            result = cached_method[pref]
         except KeyError:
-            result = self._call_remote(remote, "get_latest_package_reference", pref, headers=headers)
-            cached_method[pref] = result
+            try:
+                result = self._call_remote(remote, "get_latest_package_reference", pref,
+                                           headers=headers)
+                cached_method[pref] = result
+                return result
+            except NotFoundException as e:
+                # Let's avoid leaking memory by saving all the exception objects,
+                # which translates to a ~2x memory increase. Now, it only saves the type and the
+                # final message. For now, let's cache only the NotFoundException one.
+                cached_method[pref] = (type(e), str(e))
+                raise e
+        else:
+            if isinstance(result, tuple) and issubclass(result[0], NotFoundException):
```

Comment:
I think I'd prefer something like
```suggestion
            if isinstance(result, tuple):
```
Because if someone stores a tuple with a different exception, it will be returned naturally, not raised.
So it should be one condition only the thing that triggers the ``raise``?